### PR TITLE
Revert "RTT fixes (#7428)"

### PR DESCRIPTION
**Pull request by ClobberXD**
This reverts commit 968ce9af598024ec71e9ffb2d15c3997a13ad754.

See #8183

### CODE REVIEW

**Comment by paramat** [comment]
Thanks.
@cheapie can you test this?

**Comment by cheapie** [comment]
Doesn't compile for me:
```
/home/cheapie/source/minetest/src/network/connectionthreads.cpp: In member function ‘SharedBuffer<unsigned char> con::ConnectionReceiveThread::handlePacketType_Control(con::Channel*, SharedBuffer<unsigned char>, con::Peer*, irr::u8, bool)’:
/home/cheapie/source/minetest/src/network/connectionthreads.cpp:1158:11: warning: unused variable ‘rtt’ [-Wunused-variable]
     float rtt = 0.0f;
           ^~~
/home/cheapie/source/minetest/src/network/connectionthreads.cpp:1177:5: error: expected primary-expression before ‘catch’
   } catch (NotFoundException &e) {
     ^~~~~
/home/cheapie/source/minetest/src/network/connectionthreads.cpp:1184:9: error: expected ‘catch’ before ‘if’
  } else if (controltype == CONTROLTYPE_SET_PEER_ID) {
         ^~
/home/cheapie/source/minetest/src/network/connectionthreads.cpp:1184:9: error: expected ‘(’ before ‘if’
  } else if (controltype == CONTROLTYPE_SET_PEER_ID) {
         ^~
         (
/home/cheapie/source/minetest/src/network/connectionthreads.cpp:1184:9: error: expected type-specifier before ‘if’
  } else if (controltype == CONTROLTYPE_SET_PEER_ID) {
         ^~
/home/cheapie/source/minetest/src/network/connectionthreads.cpp:1184:8: error: expected ‘)’ before ‘if’
  } else if (controltype == CONTROLTYPE_SET_PEER_ID) {
        ^~~
        )
/home/cheapie/source/minetest/src/network/connectionthreads.cpp:1184:9: error: expected ‘{’ before ‘if’
  } else if (controltype == CONTROLTYPE_SET_PEER_ID) {
         ^~
/home/cheapie/source/minetest/src/network/connectionthreads.cpp:1227:68: error: qualified-id in declaration before ‘(’ token
 SharedBuffer<u8> ConnectionReceiveThread::handlePacketType_Original(Channel *channel,
                                                                    ^
/home/cheapie/source/minetest/src/network/connectionthreads.cpp:1241:65: error: qualified-id in declaration before ‘(’ token
 SharedBuffer<u8> ConnectionReceiveThread::handlePacketType_Split(Channel *channel,
                                                                 ^
/home/cheapie/source/minetest/src/network/connectionthreads.cpp:1272:68: error: qualified-id in declaration before ‘(’ token
 SharedBuffer<u8> ConnectionReceiveThread::handlePacketType_Reliable(Channel *channel,
                                                                    ^
/home/cheapie/source/minetest/src/network/connectionthreads.cpp: At global scope:
/home/cheapie/source/minetest/src/network/connectionthreads.cpp:1382:1: error: expected ‘}’ at end of input
 }
 ^
/home/cheapie/source/minetest/src/network/connectionthreads.cpp:29:1: note: to match this ‘{’
 {
 ^
/home/cheapie/source/minetest/src/network/connectionthreads.cpp: In member function ‘SharedBuffer<unsigned char> con::ConnectionReceiveThread::handlePacketType_Control(con::Channel*, SharedBuffer<unsigned char>, con::Peer*, irr::u8, bool)’:
/home/cheapie/source/minetest/src/network/connectionthreads.cpp:1382:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
```

**Comment by pauloue** [comment]
There's a conflict with 1d1cf000c023b77721ad8cb226330ca6a72a1acc

**Comment by paramat** [comment]
Hmm i wondered if reverting would be complicated by later changes. But still, trying to fix the bug is likely to take longer, so reverting still seems the best thing to do.

**Comment by ClobberXD** [comment]
I did encounter a conflict while reverting - I probably messed up while resolving it :)

File: src/network/connectionthreads.cpp [hunk 1]
```diff
@@ -1157,10 +1157,13 @@ SharedBuffer<u8> ConnectionReceiveThread::handlePacketType_Control(Channel *chan
 				// rtt miscalculation we handle it here
 				float rtt = 0.0f;
 				if (current_time > p.absolute_send_time) {
-					rtt = (current_time - p.absolute_send_time) / 1000.0f;
+					float rtt = (current_time - p.absolute_send_time) / 1000.0;
+
+					// Let peer calculate stuff according to it
+					// (avg_rtt and resend_timeout)
+					dynamic_cast<UDPPeer *>(peer)->reportRTT(rtt);
 				} else if (p.totaltime > 0) {
```
Comment (osjc) [inline src/network/connectionthreads.cpp]: The "{" at the end of this line shall go away as its closing "}" was obliterated by this revert.

File: src/network/connectionthreads.cpp [hunk 2]
```diff
@@ -1157,10 +1157,13 @@ SharedBuffer<u8> ConnectionReceiveThread::handlePacketType_Control(Channel *chan
...
-					rtt = p.totaltime;
-				}
```
Comment (osjc) [inline src/network/connectionthreads.cpp]: Here was the closing "{" for the "}" in line 1165.

Comment (osjc) [reply src/network/connectionthreads.cpp]: An alternative way to fix this would be to leave "deleted line 1163" in the code.

Comment (ClobberXD) [reply src/network/connectionthreads.cpp]: As you've noted, braces for one single line is against code-style, so removing `{` in L1165 is probably the way to go.

File: src/network/connectionthreads.cpp (hunk 1 above)
Comment (ClobberXD) [reply src/network/connectionthreads.cpp]: Done

File: src/network/connectionthreads.cpp [hunk 3]
```diff
@@ -1157,10 +1157,13 @@ SharedBuffer<u8> ConnectionReceiveThread::handlePacketType_Control(Channel *chan
...
-				} else if (p.totaltime > 0) {
-					rtt = p.totaltime;
-				}
+					float rtt = (current_time - p.absolute_send_time) / 1000.0;
+
+					// Let peer calculate stuff according to it
+					// (avg_rtt and resend_timeout)
+					dynamic_cast<UDPPeer *>(peer)->reportRTT(rtt);
+				} else if (p.totaltime > 0)
+					float rtt = p.totaltime;
```
Comment (pgimeno) [inline src/network/connectionthreads.cpp]: Does this line do anything at all?

**Comment by cheapie** [comment]
I can confirm that this now compiles, runs, and no longer has the original bug.

File: src/network/connectionthreads.cpp [hunk 4]
```diff
@@ -1157,10 +1157,13 @@ SharedBuffer<u8> ConnectionReceiveThread::handlePacketType_Control(Channel *chan
...
 
 				// Let peer calculate stuff according to it
```
Comment (SmallJoker) [inline src/network/connectionthreads.cpp]: Needs removing.

**Comment by VanessaE** [comment]
I've put this on my servers (at commit 8f5a716a1644d9dfa0d5bddb923a0ac25c14f446 anyway, applied on top of HEAD commit b7e1bca28c2c551198d284473c22b0293139163d, rebased).  It seems to fix the lag problem for those of my users who were affected.

File: src/network/connectionthreads.cpp (hunk 4 above)
Comment (paramat) [reply src/network/connectionthreads.cpp]: @ClobberXD

Comment (ClobberXD) [reply src/network/connectionthreads.cpp]: Done

**Comment by SmallJoker** [comment]
Your revert is still incomplete. The second statement does not do anything due to an unused local variable.
See here for the original code: https://github.com/minetest/minetest/blob/stable-0.4/src/network/connection.cpp#L2397

**Comment by ClobberXD** [comment]
Thanks for letting me know - I can't understand most of the low-level networking code. :)

**Comment by paramat** [comment]
SmallJoker will be absent until Saturday btw.

**Comment by jastevenson303** [comment]
Works but I timeout a lot; perhaps unrelated.

File: src/network/connectionthreads.cpp (hunk 3 above)
Comment (paramat) [reply src/network/connectionthreads.cpp]: Yeah should be:
`rtt = p.totaltime;`

File: src/network/connectionthreads.cpp [hunk 5]
```diff
@@ -1157,13 +1157,14 @@ SharedBuffer<u8> ConnectionReceiveThread::handlePacketType_Control(Channel *chan
 				// rtt miscalculation we handle it here
 				float rtt = 0.0f;
 				if (current_time > p.absolute_send_time) {
-					rtt = (current_time - p.absolute_send_time) / 1000.0f;
-				} else if (p.totaltime > 0) {
-					rtt = p.totaltime;
-				}
+					float rtt = (current_time - p.absolute_send_time) / 1000.0;
```
Comment (paramat) [inline src/network/connectionthreads.cpp]: `rtt = (current_time - p.absolute_send_time) / 1000.0;`
As rtt is already initialised outside the conditional (L1158).

Comment (sfan5) [reply src/network/connectionthreads.cpp]: This code is correct.
The variable in the outside scope was not there before 968ce9a, so this PR removed it again.

Comment (paramat) [reply src/network/connectionthreads.cpp]: Ok.

File: src/network/connectionthreads.cpp (hunk 3 above)
Comment (paramat) [reply src/network/connectionthreads.cpp]: Now resolved.

**Comment by paramat** [comment]
> Your revert is still incomplete. The second statement does not do anything due to an unused local variable.

Resolved by sfan5.

**Comment by paramat** [comment]
:+1: code revert looks good.

**Comment by osjc** [comment]
I would advise when re-adding the reverted commit to split it into 5 commits, one for each of the things done in it. That way when the bug tries to reappear, it can be nailed down to a smaller set of changes.

**Pull request merged**